src/jdk.incubator.vector/share/classes/jdk/incubator/vector/X-Vector.java.template

```diff
@@ -2945,6 +2958,37 @@ public abstract class $abstractvectortype$ extends AbstractVector<$Boxtype$> {
         return v.rearrange(this.toShuffle(), m);
     }
 
+
+    /**
+     * {@inheritDoc} <!--workaround-->
+     */
+    @Override
+    public abstract
+    $abstractvectortype$ selectFrom(Vector<$Boxtype$> v1, Vector<$Boxtype$> v2);
+
+
+    /*package-private*/
```

Comment:
I think we can simplify with:
```
    /*package-private*/
    @ForceInline
    final $abstractvectortype$ selectFromTemplate(Class<? extends Vector<$Boxbitstype$>> indexVecClass,
                                                  $abstractvectortype$ v1, $abstractvectortype$ v2) {
        int twoVectorLenMask = (length() << 1) - 1;
#if[FP]
        Vector<$Boxbitstype$> wrapped_indexes = this.convert(VectorOperators.{#if[intOrFloat]?F2I:D2L}, 0)
                                                   .lanewise(VectorOperators.AND, twoVectorLenMask);
        return VectorSupport.selectFromTwoVectorOp(getClass(), indexVecClass , $type$.class, $bitstype$.class,
                                                   length(), wrapped_indexes, v1, v2,
                                                   (vec1, vec2, vec3) -> selectFromTwoVectorHelper(vec1, vec2, vec3)
        );
#else[FP]
        $abstractvectortype$ wrapped_indexes = this.lanewise(VectorOperators.AND, twoVectorLenMask);
        return VectorSupport.selectFromTwoVectorOp(getClass(), indexVecClass, $type$.class, $type$.class,
                                                   length(), wrapped_indexes, v1, v2,
                                                   (vec1, vec2, vec3) -> selectFromTwoVectorHelper(vec1, vec2, vec3)
        );
#end[FP]
    }
```
(Note that's without the assert - see separate comment).